File: w09/d01/amy_ruan/mailer/spec/mailbox_spec.rb
require_relative '../lib/mailbox'
require_relative '../lib/package'
require_relative '../lib/postcard'

describe Mailbox do
  it "has an array of Mailitem objects" do
    mailbox = Mailbox.new
    expect(mailbox.contents.class).to eq(Array)
  end

  it "should have an accept mail method" do
    mailbox= Mailbox.new
    mailbox.accept_mail("hello")
    expect(mailbox.contents.empty?).to eq(false)
  end

  it "should have an open mail method" do
    mailbox = Mailbox.new
    mailbox.accept_mail("hello")
    mailbox.accept_mail("byebye")
    expect(mailbox.open).to eq("hello")
    expect(mailbox.contents.length).to eq(1)
  end

  it "should have a to_s method that
  includes the number of postcards and
  packages in the mailbox" do
    mailbox = Mailbox.new
    package = Package.new
    postcard = Postcard.new
    mailbox.accept_mail(package)
    mailbox.accept_mail(postcard)
    expect(mailbox.to_s).to eq("The mailbox has 1 postcard and 1 package")
  end


end

File: w09/d03/jorge_colindres/mail_objects_app/spec/javascripts/mailbox_spec.js
describe('a mailbox', function(){

  var mailbox, postcard, packageObject;
  beforeEach(function(){
    mailbox = new Mailbox();
    postcard = new Postcard('hello');
    packageObject = new Package('book');
  });

  it('should have an array of mail item objects', function(){
    expect(mailbox.mailitems).toBeDefined();
    mailbox.acceptMail(postcard);
    mailbox.acceptMail(packageObject);
    expect(mailbox.mailitems).toContain(postcard);
    expect(mailbox.mailitems.length).toEqual(2);
  });

  it('should have an acceptMail method', function(){
    expect(mailbox.acceptMail).toBeDefined();
  });

  it('should accept postcards and packages and add them to the mailitems array', function(){
    mailbox.acceptMail(postcard);
    mailbox.acceptMail(packageObject);
    expect(mailbox.mailitems).toContain(postcard);
  });

  it('should have an openMail method that removes and returns the first item in mailitems', function(){
    mailbox.acceptMail(postcard);
    mailbox.acceptMail(packageObject);
    mailbox.openMail();
    expect(mailbox.mailitems.length).toEqual(1);
    expect(mailbox.mailitems).not.toContain(postcard);
  });

  it('should have a toString method that includes the number of postcards and packages in it', function(){
    mailbox.acceptMail(postcard);
    mailbox.acceptMail(packageObject);
    expect(mailbox.toString()).toEqual('The mailbox has 1 postcards and 1 packages.');
  });
});
